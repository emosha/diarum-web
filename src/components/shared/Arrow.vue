<template>
  <div class="arrow">
    <a href="#" v-scroll-to="href" class="arrow__link">
      <div class="arrow__container">
        <font-awesome-icon :icon="icon"/>
      </div>
    </a>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component

export default class Arrow extends Vue {
  @Prop() public href: string|undefined;
  @Prop() public direction: string|undefined;
  public icon = `chevron-${this.direction}`;
}
</script>

<style scoped lang="scss">
@keyframes jump {
  0% {
    top: 0px;
  }
  50% {
    top: 30px;
  }
  100% {
    top: 0px;
  }
}

.arrow {
  &__container {
    position: relative;
    animation: jump 3s infinite;

    .svg-inline--fa {
      font-size: 40px;
      color: #5d59cb;
    }
  }
}
</style>
