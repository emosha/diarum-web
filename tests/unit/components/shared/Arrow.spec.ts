import { shallowMount } from '@vue/test-utils';
import Arrow from '@/components/shared/Arrow.vue';

describe('Arrow.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(Arrow, {
      propsData: {
        direction: 'up',
        href: '#features',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
