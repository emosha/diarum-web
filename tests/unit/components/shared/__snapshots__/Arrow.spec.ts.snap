// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow.vue renders correctly 1`] = `
<div class="arrow"><a href="#" class="arrow__link">
    <div class="arrow__container">
      <font-awesome-icon icon="chevron-up"></font-awesome-icon>
    </div>
  </a></div>
`;
